tools/spectral/ipa/rulesets/functions/eachCustomMethodMustBeGetOrPost.js

```diff
@@ -0,0 +1,33 @@
+import { isCustomMethod } from './utils/resourceEvaluation.js';
+
+const ERROR_MESSAGE = 'The HTTP method for custom methods must be GET or POST.';
+const VALID_METHODS = ['get', 'post'];
+
+export default (paths) => {
+  // Collect all errors
+  const errors = [];
+
+  // Iterate through each path key and its corresponding path item
+  for (const [pathKey, pathItem] of Object.entries(paths)) {
```

Comment:
Ah didn't realize that `paths` is not an array, I thought that the change would only make the input contain a single path and it's pathitems, not the whole `paths` object. Sorry about that! This makes the function a bit harder to read IMO, feel free to revert back to the previous approach if you want. WDYT?

Reply:
Or perhaps try `$.paths[*]`

Reply:
It worked with `$.paths[*]` and using `path` from the `context`. Could you take another look and let me know how it looks?

Reply:
`$.paths[*]` gives you the object for each path, but it doesn't include the path keys themselves, so it's not quite enough on its own

Reply:
Thanks! Looks good! Just one question on the logic